lib/cog_api/http/roles.ex

```diff
@@ -7,8 +7,8 @@ defmodule CogApi.HTTP.Roles do
     Base.get(endpoint, "roles") |> Base.format_response("roles", [%Role{}])
   end
 
-  def show(%Endpoint{}=endpoint, id) do
-    Base.get(endpoint, "roles/#{id}") |> Base.format_response("role", %Role{})
+  def show(%Endpoint{}=endpoint, role_name) do
+    Base.get_by(endpoint, "roles", name: role_name) |> Base.format_response("role", %Role{})
   end
```

Comment:
Not a fan of this, but after discussion with team about it, it was decided that performing a call to the controller passing the role name would have to wait a little.